Px.Utils/Validation/SyntaxValidation/SyntaxValidationUtilityMethods.cs

```diff
@@ -384,6 +399,137 @@ internal static int GetLineChangesValidity(string value, PxFileSyntaxConf syntax
+                return false;
+            }
+
+            // Time value series should be a list of time values
+            string[] timeValSeries = input.Split(syntaxConf.Symbols.Value.ListSeparator);
+            if (Array.TrueForAll(timeValSeries, x => IsValidTimestampFormat(x, timeInterval, syntaxConf)))
+            {
+                valueFormat = ValueType.TimeValSeries;
+                return true;
+            }
+            else
+            {
+                valueFormat = null;
+                return false;
+            }
+        }
+
+        private static bool IsValidTimestampFormat(string input, string timeInterval, PxFileSyntaxConf syntaxConf)
+        {
+            // Timestamp should match with a regex pattern for the given time interval
+            input = CleanString(input, syntaxConf, true);
+            Regex regex = GetRegexForTimeInterval(timeInterval, syntaxConf);
+
+            try
+            {
+                return regex.IsMatch(input);
+            }
+            catch (RegexMatchTimeoutException)
+            {
+                // If the regex times out, we can't be sure if the input is a valid timestamp
+                return false;
+            }
+        }
+
+        private static Regex GetRegexForTimeInterval(string timeInterval, PxFileSyntaxConf syntaxConf)
+        {
+            if (!regexPatterns.TryGetValue(timeInterval, out Regex? regex))
+            {
+                string pattern = GetPatternForTimeInterval(timeInterval, syntaxConf);
+                regex = new Regex(pattern, RegexOptions.Compiled | RegexOptions.Singleline, timeout);
+                regexPatterns[timeInterval] = regex;
+            }
+
+            return regex;
+        }
+
+        private static string GetPatternForTimeInterval(string timeInterval, PxFileSyntaxConf syntaxConf)
+        {
+            return timeInterval switch
+            {
+                string interval when interval == syntaxConf.Tokens.Time.YearInterval => @"^\d{4}$",
+                string interval when interval == syntaxConf.Tokens.Time.HalfYearInterval => @"^\d{4}[1-4]$",
+                string interval when interval == syntaxConf.Tokens.Time.TrimesterInterval => @"^\d{4}[1-4]$",
+                string interval when interval == syntaxConf.Tokens.Time.QuarterYearInterval => @"^\d{4}[1-4]$",
+                string interval when interval == syntaxConf.Tokens.Time.MonthInterval => @"^\d{4}(0[1-9]|1[0-2])$",
+                string interval when interval == syntaxConf.Tokens.Time.WeekInterval => @"^\d{4}(0[1-9]|[1-4][0-9]|5[0-2])$",
+                string interval when interval == syntaxConf.Tokens.Time.DayInterval => @"^\d{4}(0[1-9]|1[0-2])(0[1-9]|[1-2][0-9]|3[0-1])$",
+                _ => throw new ArgumentException("Invalid time interval", timeInterval)
+            };
+        }
+
+        private static bool TryGetTimeValueRangeFormat(string input, string timeInterval, out ValueType? valueFormat, PxFileSyntaxConf syntaxConf)
```

Comment:
The return value of this function is never used, so the Try structure should be changed to a normal method.